transports/quic/src/webtransport/fingerprint.rs

```diff
@@ -0,0 +1,77 @@
+use sha2::Digest as _;
+use std::fmt;
+
+const SHA256: &str = "sha-256";
+const MULTIHASH_SHA256_CODE: u64 = 0x12;
+
+type Multihash = libp2p_core::multihash::Multihash<64>;
+
+/// A certificate fingerprint that is assumed to be created using the SHA256 hash algorithm.
+#[derive(Eq, PartialEq, Copy, Clone)]
+pub struct Fingerprint([u8; 32]);
```

Comment:
I am wondering if we should eventually upstream (parts) of this to `rust-multiaddr` and have `Protocol::Certhash` take a `Fingerprint`.